third-party/realdds/include/realdds/dds-device-impl.h

```diff
@@ -206,117 +236,106 @@ class dds_device::impl
                         case topics::device::notification::msg_type::DEVICE_HEADER:
                             if( state_type::WAIT_FOR_DEVICE_HEADER == state )
                             {
-                                _num_of_sensors
-                                    = data.get< topics::device::notification::device_header_msg >()->num_of_sensors;
-                                LOG_INFO( "got DEVICE_HEADER message with " << _num_of_sensors << " sensors" );
-                                state = state_type::WAIT_FOR_SENSOR_HEADER;
-                            }
-                            else
-                                LOG_ERROR( "Wrong message received, got 'DEVICE_HEADER' when the state is: " << state );
-                            break;
-                        case topics::device::notification::msg_type::SENSOR_HEADER:
+                                auto msg = data.get< topics::device::notification::device_header_msg >();
+                                if ( msg  == nullptr )
+                                {
+                                    throw std::runtime_error( "get< topics::device::notification::device_header_msg > return nullptr" );
```

Comment:
DDS_THROW

Reply:
Actually, never mind -- I changed this same code, doing the same thing -- I'll do it.